Test flake

One of the tests seems to fail randomly quite often

```python
FAILED tests/test_appclass.py::test_dependencies - traitlets.traitlets.TraitError: Version v22.11.1 is not available for quantum-espresso app.
```

CC @unkcpz

--- a/aiidalab/utils.py
+++ b/aiidalab/utils.py
@@ -33,27 +33,27 @@ FIND_INSTALLED_PACKAGES_CACHE = TTLCache(maxsize=32, ttl=60)  # type: ignore
 # The cache is configured to avoid spamming the app registry server with requests
 # that are made in rapid succession and also serves as a fallback in case
 # that the index server is temporarily not reachable.
-session = CachedSession(
+_session = CachedSession(
     "aiidalab_registry",
     use_cache_dir=True,  # store cache in ~/.cache/
     backend="sqlite",
     expire_after=60,  # seconds
-    old_data_on_error=True,
+    stale_if_error=True,
 )
 
 
 def load_app_registry_index() -> Any:
     """Load apps' information from the AiiDAlab registry."""
     try:
-        return session.get(f"{AIIDALAB_REGISTRY}/apps_index.json").json()
+        return _session.get(f"{AIIDALAB_REGISTRY}/apps_index.json").json()
     except (ValueError, requests.ConnectionError) as error:
         raise RuntimeError("Unable to load registry index") from error
 
 
 def load_app_registry_entry(app_id: str) -> Any:
-    """Load registry enty for app with app_id."""
+    """Load registry entry for app with app_id."""
     try:
-        return session.get(f"{AIIDALAB_REGISTRY}/apps/{app_id}.json").json()
+        return _session.get(f"{AIIDALAB_REGISTRY}/apps/{app_id}.json").json()
     except (ValueError, requests.ConnectionError):
         logger.debug(f"Unable to load registry entry for app with id '{app_id}'.")
         return None

--- a/aiidalab/utils.py
+++ b/aiidalab/utils.py
@@ -20,6 +20,7 @@ import requests
 from cachetools import TTLCache, cached
 from packaging.requirements import Requirement
 from packaging.utils import NormalizedName, canonicalize_name
+from requests_cache import CachedSession
 
 from .config import AIIDALAB_REGISTRY
 from .environment import Environment
@@ -29,37 +30,22 @@ from .metadata import Metadata
 logger = logging.getLogger(__name__)
 FIND_INSTALLED_PACKAGES_CACHE = TTLCache(maxsize=32, ttl=60)  # type: ignore
 
-# Warning: try-except is a fix for Quantum Mobile release v19.03.0 where
-# requests_cache is not installed.
-try:
-    import requests_cache
-except ImportError:
-    logger.warning(
-        "The requests_cache package is missing.  Requests made to the app "
-        "registry will not be cached!"
-    )
-else:
-    # Install global cache for all requests.
-
-    # The cache file is placed within the user home directory.
-    cache_file = Path.home().joinpath(".cache", "requests", "cache")
-    cache_file.parent.mkdir(parents=True, exist_ok=True)
-
-    # The cache is configured to avoid spamming the index server with requests
-    # that are made in rapid succession and also serves as a fallback in case
-    # that the index server is not reachable.
-    requests_cache.install_cache(
-        cache_name=str(cache_file),
-        backend="sqlite",
-        expire_after=60,  # seconds
-        old_data_on_error=True,
-    )
+# The cache is configured to avoid spamming the app registry server with requests
+# that are made in rapid succession and also serves as a fallback in case
+# that the index server is temporarily not reachable.
+session = CachedSession(
+    "aiidalab_registry",
+    use_cache_dir=True,  # store cache in ~/.cache/
+    backend="sqlite",
+    expire_after=60,  # seconds
+    old_data_on_error=True,
+)
 
 
 def load_app_registry_index() -> Any:
     """Load apps' information from the AiiDAlab registry."""
     try:
-        return requests.get(f"{AIIDALAB_REGISTRY}/apps_index.json").json()
+        return session.get(f"{AIIDALAB_REGISTRY}/apps_index.json").json()
     except (ValueError, requests.ConnectionError) as error:
         raise RuntimeError("Unable to load registry index") from error
 
@@ -67,7 +53,7 @@ def load_app_registry_index() -> Any:
 def load_app_registry_entry(app_id: str) -> Any:
     """Load registry enty for app with app_id."""
     try:
-        return requests.get(f"{AIIDALAB_REGISTRY}/apps/{app_id}.json").json()
+        return session.get(f"{AIIDALAB_REGISTRY}/apps/{app_id}.json").json()
     except (ValueError, requests.ConnectionError):
         logger.debug(f"Unable to load registry entry for app with id '{app_id}'.")
         return None

--- a/aiidalab/utils.py
+++ b/aiidalab/utils.py
@@ -20,7 +20,6 @@ import requests
 from cachetools import TTLCache, cached
 from packaging.requirements import Requirement
 from packaging.utils import NormalizedName, canonicalize_name
-from requests_cache import CachedSession
 
 from .config import AIIDALAB_REGISTRY
 from .environment import Environment
@@ -30,16 +29,27 @@ from .metadata import Metadata
 logger = logging.getLogger(__name__)
 FIND_INSTALLED_PACKAGES_CACHE = TTLCache(maxsize=32, ttl=60)  # type: ignore
 
-# The cache is configured to avoid spamming the app registry server with requests
-# that are made in rapid succession and also serves as a fallback in case
-# that the index server is temporarily not reachable.
-_session = CachedSession(
-    "aiidalab_registry",
-    use_cache_dir=True,  # store cache in ~/.cache/
-    backend="sqlite",
-    expire_after=60,  # seconds
-    stale_if_error=True,
-)
+# NOTE: try-except is a fix for Quantum Mobile release v19.03.0 where
+# requests_cache is not installed.
+try:
+    # The cache is configured to avoid spamming the app registry server with requests
+    # that are made in rapid succession and also serves as a fallback in case
+    # that the index server is temporarily not reachable.
+    from requests_cache import CachedSession
+
+    _session = CachedSession(
+        "aiidalab_registry",
+        use_cache_dir=True,  # store cache in ~/.cache/
+        backend="sqlite",
+        expire_after=60,  # seconds
+        stale_if_error=True,
+    )
+except ImportError:
+    logger.warning(
+        "The requests_cache package is missing. "
+        "Requests made to the app registry will not be cached."
+    )
+    _session = requests.Session()
 
 
 def load_app_registry_index() -> Any:

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -8,7 +8,7 @@ from ruamel.yaml import YAML
 def static_path():
     import importlib.resources
 
-    return importlib.resources.files() / "static"
+    return importlib.resources.files(__name__) / "static"
 
 
 @pytest.fixture(scope="session")

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,18 +1,14 @@
 import sys
-from pathlib import Path
 
 import pytest
 from ruamel.yaml import YAML
 
-from aiidalab.app import AiidaLabApp, _AiidaLabApp
-
 
 @pytest.fixture(scope="session")
 def static_path():
-    # TODO: Switch to importlib.resources once we drop support for Python 3.8
-    # import importlib.resources
-    # return importlib.resources.files() / "static"
-    return Path(__file__).parent.absolute() / "static"
+    import importlib.resources
+
+    return importlib.resources.files() / "static"
 
 
 @pytest.fixture(scope="session")
@@ -40,6 +36,8 @@ def generate_app(monkeypatch, app_registry_path):
         app_data=None,
         watch=False,
     ):
+        from aiidalab.app import AiidaLabApp, _AiidaLabApp
+
         if app_data is None:
             safe_yaml = YAML(typ="safe")
             with app_registry_path.open() as f:
@@ -49,6 +47,7 @@ def generate_app(monkeypatch, app_registry_path):
         # it is a installed app. Following monkeypatch make it more close
         # to the real scenario for test.
         monkeypatch.setattr(_AiidaLabApp, "is_installed", lambda _: True)
+        monkeypatch.setattr(_AiidaLabApp, "is_registered", lambda _: True)
         app = AiidaLabApp(name, app_data, aiidalab_apps_path, watch=watch)
 
         return app
@@ -90,13 +89,10 @@ def forbid_external_commands(monkeypatch):
         lambda: raise_exc("Running `verdi daemon restart` not allowed in tests!"),
     )
 
-    # TODO: Enable this once the existing tests don't use it
-    """
     monkeypatch.setattr(
         "aiidalab.utils.load_app_registry_index",
-        lambda: raise_exc("Running fetching registry index not allowed in tests!"),
+        lambda: raise_exc("Fetching registry index not allowed in tests!"),
     )
-    """
 
 
 @pytest.fixture

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -93,6 +93,11 @@ def forbid_external_commands(monkeypatch):
         lambda: raise_exc("Fetching registry index not allowed in tests!"),
     )
 
+    monkeypatch.setattr(
+        "aiidalab.utils.load_app_registry_entry",
+        lambda _: raise_exc("Fetching app registry entry not allowed in tests!"),
+    )
+
 
 @pytest.fixture
 def installed_packages(monkeypatch):

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,4 +1,5 @@
 import sys
+from pathlib import Path
 
 import pytest
 from ruamel.yaml import YAML
@@ -6,9 +7,7 @@ from ruamel.yaml import YAML
 
 @pytest.fixture(scope="session")
 def static_path():
-    import importlib.resources
-
-    return importlib.resources.files(__name__) / "static"
+    return Path(__file__).parent.absolute() / "static"
 
 
 @pytest.fixture(scope="session")

--- a/tests/test_appclass.py
+++ b/tests/test_appclass.py
@@ -6,7 +6,7 @@ from time import sleep
 import pytest
 import traitlets
 
-from aiidalab.app import AiidaLabApp, AiidaLabAppWatch, _AiidaLabApp
+from aiidalab.app import AiidaLabApp, AiidaLabAppWatch
 
 
 def test_init_refresh(generate_app):
@@ -45,10 +45,10 @@ def test_dependencies(generate_app):
 @pytest.mark.usefixtures("installed_packages")
 def test_app_is_not_registered(generate_app, monkeypatch, tmp_path):
     """test the app is not registered and the available versions are empty."""
-    # monkeypatch and make the app not registered
-    monkeypatch.setattr(_AiidaLabApp, "is_registered", lambda _: False)
 
     app: AiidaLabApp = generate_app()
+    # monkeypatch and make the app not registered
+    monkeypatch.setattr(app._app, "is_registered", lambda: False)
     app.refresh()
 
     assert app.is_installed() is True

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -103,10 +103,15 @@ def test_dry_install(aiidalab_env, static_path):
     assert "No apps installed" in result.output, result.output
 
 
-def test_install_uninstall(aiidalab_env, static_path):
+def test_install_uninstall(monkeypatch, aiidalab_env, static_path):
     """
     Test `aiidalab install` followed by `aiidalab uninstall`.
     """
+    monkeypatch.setattr(
+        "aiidalab.utils.load_app_registry_entry",
+        lambda _: None,
+    )
+
     aiidalab_apps = Path(aiidalab_env["AIIDALAB_APPS"])
     app_name = "hello-world"
     app_path = static_path / "app_with_setupcfg"

--- a/tests/test_conftest.py
+++ b/tests/test_conftest.py
@@ -8,6 +8,8 @@ def test_forbidden_functions():
     and other functions that might influence global environment.
     """
     from aiidalab.utils import (
+        load_app_registry_entry,
+        load_app_registry_index,
         run_pip_install,
         run_post_install_script,
         run_verdi_daemon_restart,
@@ -24,3 +26,9 @@ def test_forbidden_functions():
     with pytest.raises(SystemExit):
         process = run_post_install_script("post_install")
         process.wait()
+
+    with pytest.raises(SystemExit):
+        load_app_registry_index()
+
+    with pytest.raises(SystemExit):
+        load_app_registry_entry("app")

